hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -93,18 +93,14 @@ public HoodieMergeHandle(HoodieWriteConfig config, String instantTime, HoodieTab
     init(fileId, this.partitionPath, dataFileToBeMerged);
   }
 
-  @Override
-  public Schema getWriterSchemaWithMetafields() {
-    return writerSchemaWithMetafields;
-  }
-
   public Schema getWriterSchema() {
-    return writerSchema;
+    return tableSchema;
```

Comment:
Feel we could rename this method to getReaderSchema() since this is used only at one place 
```
bootstrapReadSchema = mergeHandle.getWriterSchema();
```
And we assign it to a var to store read schema anyways.

Reply:
It seems reasonable!